### PR TITLE
FCCD Models refactor

### PR DESCRIPTION
**Pull request by isaaccorley**
This PR;
- removes `FCEF` model since this is basically a U-Net with the 2 change images stacked together so the input channels are 2*num_channels.
- Refactors `FCSiamConc` and `FCSiamDiff` to inherit from `segmentation_models_pytorch.Unet`. These models are basically U-Nets but feed both images through the U-Net encoder and then either take the difference (`FCSiamDiff`) or concatenate (`FCSiamConc`).

Notes:
- By inheriting from smp this allows us to easily load different encoder backbone weights (I'll add a follow up PR to add this later). Previously the backbone was fixed.
- For the models I'm reusing the `smp.Unet` constructor and it's args. For `FCSiamDiff` I made a dummy constructor so that I had a place to put the docstring. Not sure if this is the best way.

### CODE REVIEW

**Comment by isaaccorley** [comment]
@adamjstewart I'm getting the below errors in the docs. Do you know if there's a quick fix for these? I'm basically inheriting from some segmentation_models_pytorch classes and I think it's unable to find information from the parent classes.

```
generating indices... /home/docs/checkouts/readthedocs.org/user_builds/torchgeo/checkouts/345/torchgeo/models/__init__.py:docstring of torchgeo.models.FCSiamConc:1: WARNING: py:class reference target not found: segmentation_models_pytorch.base.model.SegmentationModel
/home/docs/checkouts/readthedocs.org/user_builds/torchgeo/checkouts/345/torchgeo/models/__init__.py:docstring of torchgeo.models.FCSiamDiff:1: WARNING: py:class reference target not found: segmentation_models_pytorch.unet.model.Unet
```

**Comment by adamjstewart** [comment]
@isaaccorley you need to add smp to the `intersphinx_mapping` in `docs/conf.py`.

**Comment by isaaccorley** [comment]
Still can't seem to get this one to work. Wonder if it's similar to your issue with the PyTorch modules.

**Review by calebrob6 — APPROVED** [review]

File: docs/conf.py [hunk 1]
```diff
@@ -61,6 +61,8 @@
     ("py:class", "LightningDataModule"),
     # Undocumented class
     ("py:class", "torchvision.models.resnet.ResNet"),
+    ("py:class", "segmentation_models_pytorch.base.model.SegmentationModel"),
+    ("py:class", "segmentation_models_pytorch.unet.model.Unet"),
```
Comment (adamjstewart) [inline docs/conf.py:65]: Unet is documented: https://segmentation-modelspytorch.readthedocs.io/en/latest/docs/api.html#unet

Might need to change `__module__` though. SegmentationModel isn't documented as far as I can tell.

**Comment by calebrob6** [comment]
Cool should be good now

**Review by calebrob6 — APPROVED** [review]

**Comment by adamjstewart** [comment]
When I make the 0.3.0 release notes, remind me to add this FCEF removal as a backward incompatible change in case anyone was relying on it.

Also, who was it who added these models? Should we ping them to review? (guess we should've done that before merging but it's not too late to undo anything if it's problematic)

**Pull request merged**